extensions/smallrye-reactive-messaging-kafka/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/DefaultSerdeDiscoveryState.java

```diff
@@ -0,0 +1,143 @@
+package io.quarkus.smallrye.reactivemessaging.kafka.deployment;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.Type;
+
+import io.smallrye.reactive.messaging.kafka.KafkaConnector;
+
+class DefaultSerdeDiscoveryState {
+    private final IndexView index;
+
+    private final Map<String, Boolean> isKafkaConnector = new HashMap<>();
+
+    private Boolean hasConfluent;
```

Comment:
any reason to use objects?

Reply:
It's a cache. Goes from unknown (`null`) to either `true` or `false`.